near/omni-bridge/src/lib.rs

```diff
@@ -2217,14 +2268,28 @@ impl Contract {
     }
 
     fn resolve_utxo_fin_transfer_internal(
+        &mut self,
         token_id: AccountId,
         amount: U128,
         utxo_fin_transfer_msg: UtxoFinTransferMsg,
+        origin_chain: ChainKind,
+        storage_owner: &AccountId,
     ) -> U128 {
         let is_ft_transfer_call = !utxo_fin_transfer_msg.msg.is_empty();
         if Self::is_refund_required(is_ft_transfer_call) {
             amount
         } else {
+            let required_storage_balance = self.add_fin_utxo_transfer(&UnifiedTransferId {
```

Comment:
We have to add the fin transfer before transferring the tokens, otherwise the fin utxo transfer will be reverted but the recipent received the tokens